### PR TITLE
feat(TDKN-167):remove refreshProperties and added ref after callback

### PR DESCRIPTION
**Pull request by sgandon**
**What is the current behavior?** (You can also link to an open issue here)
rest api and salesforce had a refreshProperties method, a kind of callback that was created has created especially for DP to have some properties initialized after a datastore is set.

**What is the new behavior?**
There is no more custom method but rather using a standard callback (trigger) concept for this behavior.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
depends on https://github.com/Talend/daikon/pull/246

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/607/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/607/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/610/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/610/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/613/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/613/)

File: services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleComponentTestIT.java [hunk 1]
```diff
@@ -119,4 +119,26 @@ private ObjectNode getFileAsObjectNode(String file) throws java.io.IOException {
...
+                .accept(APPLICATION_JSON_UTF8_VALUE) //
+                .expect().statusCode(200).log().ifError() //
+                .post("properties/{definitionName}", dataSetDefinitionName);
+
+        // then
+        ObjectNode fullexampleProperties = mapper.readerFor(ObjectNode.class).readValue(response.asInputStream());
+        assertNotNull(fullexampleProperties.get("jsonSchema"));
+        assertNotNull(fullexampleProperties.get("properties"));
+        assertNotNull(fullexampleProperties.get("uiSchema"));
+        assertEquals("bar", fullexampleProperties.get("properties").get("testAfterDatastoreTrigger").textValue());
```
Comment (gonchar-ivan) [inline services/components-api-service-rest/src/test/java/org/talend/components/service/rest/fullexample/FullExampleComponentTestIT.java]: You may use RestAssured library API here:

1. It has given(), when(), then() methods. You may try to use it instead comments

2. you may call something like 
`.body("jsonSchema", notNull());`
after .post() method

3. You may use
`.body("properties.testAfterDatastoreTrigger", equalTo("bar"))`

Several calls successively are allowed: `body().body().body()` . 
Another one method, which could be used in `content()`.

See, https://github.com/Talend/components/blob/igonchar/TCOMP-558_it/services/components-api-service-rest-integration/src/test/java/org/talend/components/service/rest/JdbcDatasetTestIT.java#L176-L185

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/621/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/621/)

**Pull request merged**